tests/cuda_interop/cuda_interop_device_function.cu

```diff
@@ -0,0 +1,90 @@
+/*******************************************************************************
+//
+//  SYCL 2020 Conformance Test Suite
+//
+//  Copyright (c) 2022 The Khronos Group Inc.
+//
+*******************************************************************************/
+
+#include "cuda_helper.hpp"
+
+using namespace sycl_cts::util;
+
+template <typename T>
+SYCL_EXTERNAL __device__ T func(T *i) {
+  return *i + 1;
+}
+
+/** check device function can be called with the backend interop type `T *`
+converted from an `accessor`
+  */
+template <typename T>
+void test_device_function_buffer(sycl::queue &queue,
+                                 const std::string &typeName) {
+  size_t constexpr size = 1;
+  T data[size];
+  data[0] = 0;
+  bool result[size] = {false};
+  {
+    sycl::buffer<T> buff(data, sycl::range<1>(size));
+    sycl::buffer<bool> result_buf(result, sycl::range<1>(size));
+
+    queue.submit([&](sycl::handler &cgh) {
+      auto acc = buff.template get_access<sycl::access::mode::read>(cgh);
+      auto result_acc = result_buf.get_access<sycl::access::mode::write>(cgh);
+
+      cgh.single_task([=] {
+        result_acc[0] =
+            func<T>(sycl::get_native<sycl::backend::cuda>(acc)) == 1;
+      });
+    });
+  }
+
+  INFO("Checking for CUDA device function interop with accessor for \"" +
+       typeName + "\" type");
+  CHECK(result[0]);
+}
+
+/** check device function can be called with the backend interop type `T *`
+converted from an `local_accessor`
+  */
+template <typename T>
+void test_device_function_local(sycl::queue &queue,
+                                const std::string &typeName) {
+  size_t constexpr size = 1;
+  bool result[size] = {false};
+  {
+    sycl::buffer<bool> result_buf(result, sycl::range<1>(size));
+
+    queue.submit([&](sycl::handler &cgh) {
+      sycl::local_accessor<T> acc(size, cgh);
+      auto result_acc = result_buf.get_access<sycl::access::mode::write>(cgh);
+
+      cgh.single_task([=] {
```

Comment:
From the [spec](https://www.khronos.org/registry/SYCL/specs/sycl-2020/html/sycl-2020.html#sec:accessor.local):
>A local accessor must not be used in a [SYCL kernel function](https://www.khronos.org/registry/SYCL/specs/sycl-2020/html/sycl-2020.html#sycl-kernel-function) that is invoked via single_task or via the simple form of parallel_for that takes a range parameter. In these cases submitting the kernel to a queue must throw a synchronous exception with the errc::kernel_argument error code.

I think we should change `single_task` to `parallel_for_work_group`.

Reply:
Agreed, I have changed the local accessor checks in `cuda_interop_device_function.cu` and `cuda_interop_kernel_func_tests.hpp` to use `parallel_for_work_group`.